xain_fl/coordinator/metrics_store.py

```diff
@@ -0,0 +1,127 @@
+"""XAIN FL Metric Store"""
+
+from abc import ABC, abstractmethod
+from datetime import datetime, timedelta
+from typing import Dict, List
+
+from influxdb import InfluxDBClient
+from numpy import array2string, ndarray
+
+from xain_fl.config import MetricsConfig
+from xain_fl.logger import StructLogger, get_logger
+
+logger: StructLogger = get_logger(__name__)
+
+
+class AbstractMetricsStore(ABC):  # pylint: disable=too-few-public-methods
+    """An abstract metric store."""
+
+    @abstractmethod
+    def write_metrics(self, participant_id: str, metrics: Dict[str, ndarray]) -> bool:
+        """Write the participant metrics on behalf of the participant with the given participant_id
+        into a metric store.
+
+        Args:
+
+            participant_id: The ID of the participant.
+            metrics: The metrics of the participant with the given participant_id.
+
+        Returns:
+
+            True, on success, otherwise False.
+        """
+
+
+class NullObjectMetricsStore(AbstractMetricsStore):  # pylint: disable=too-few-public-methods
+    """A metric store that does nothing."""
+
+    def write_metrics(self, participant_id: str, metrics: Dict[str, ndarray]) -> bool:
+        """A method that has no effect.
+
+        Args:
+
+            participant_id: The ID of the participant.
+            metrics: The metrics of the participant with the given participant_id.
+
+        Returns:
+
+            Always True.
+        """
+        return True
+
+
+class MetricsStore(AbstractMetricsStore):  # pylint: disable=too-few-public-methods
+    """A metric store that uses InfluxDB to store the metrics."""
+
+    def __init__(self, config: MetricsConfig):
+        self.config = config
+        self.influx_client = InfluxDBClient(
+            self.config.host,
+            self.config.port,
+            self.config.user,
+            self.config.password,
+            self.config.db_name,
+        )
+
+    def write_metrics(self, participant_id: str, metrics: Dict[str, ndarray]) -> bool:
+        """Write the participant metrics on behalf of the participant with the given participant_id
+        into InfluxDB.
+
+        Args:
+
+            participant_id: The ID of the participant.
+            metrics: The metrics of the participant with the given participant_id.
+
+        Returns:
+
+            True, on success, otherwise False.
+        """
+
+        # FIXME: We will change the data format of the metrics message in a separate ticket.
+        # The goal is, that coordinator doesn't need to transform the metrics anymore.
```

Comment:
I'm not sure to understand this comment: does that means that the `metrics` argument will be a dictionary with the right structure and that the transformation will be done by the caller?

Reply:
Right. The idea is, that a participant sends the metrics already in s influx compatible data schema so that the coordinator just has to pass the data to InfluxDB without any transformation of the data.